app/services/order_details/reconciler.rb

```diff
@@ -102,10 +82,34 @@ def full_errors
 
     private
 
+    def load_selected_order_details
+      return [] if @params.blank?
+
+      selected_ids = @params.select { |_, p| p[:selected] == "1" }.keys
+
+      selected_ids.any? ? @order_detail_scope.where(id: selected_ids) : []
```

Comment:
nit: I think `@order_detail_scope.where(id: selected_ids)` should retrieve [] when there are no selected ids